### ISSUE TITLE
Configuration server

### ISSUE THREAD

**Issue by ChristopherRogers1991** [MycroftAI/mycroft-core#368]
Would be nice to have a simple way to enable/disable and configure skills. I think the best option would be a webpage, since that covers everything (Android, iOS, Linux, Win, Mac...).

I have a VERY DIRTY proof of concept [here](https://github.com/ChristopherRogers1991/mycroft-core/tree/feature/config_server) that allows enabling/disabling skills (start mycroft, then run the `config_server.py` file in `mycroft/configuration` and navigate to [http://localhost:8080](http://localhost:8080) if you want to test). The vision is to have something similar, where the skill names link to a page specific to that skill. Skills could provide an HTML page (or snippet/template) that gives information about themselves (e.g. sample phrases). A form could be added to these pages to allow manipulation of the config options for the skill (i.e. surface the .ini info, and allow for changing it).

My web dev skills are pretty rusty and it's never been my focus, but I'm happy to continue working on this if there is interest, and it's not already in the works elsewhere.

**Comment by penrods** [comment]
We are adding Skill configuration to the Skill API and web backend in the near future, but I can understand if you want to do this as a stopgap until that is available.  Once our work is complete, you will be able to configure the installed Skills for all devices and set all options for those Skills (either for just the single device or for all devices you have registered on your account).  The Skill writer will just need to define the options with some metadata, e.g.: option name, type (number, string, password, bool, etc), label (to show in the UI), etc.

**Comment by ChristopherRogers1991** [comment]
Ah, cool. I don't think there's any sense in duplicating effort, so if there's already something cooking, I'll just close this out.

Is the work you described being tracked on an issue, or anything I could follow?
